errors on changing theme

when changing in-app theme from light to dark the app force close, probably also other fixes needed here and there

on the main screen, just changing theme raise a runtime exception
I'm on android 8.1 here is what it says on my phone:

`java.lang.RuntimeException: Unable to start activity ComponentInfo{com.chanapps.four.activity/com.chanapps.four.activity.BoardSelectorActivity}: java.lang.IllegalArgumentException: Cannot add a null child view to a ViewGroup
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2895)
 at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4766)
 at android.app.ActivityThread.-wrap18(Unknown Source:0)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loop(Looper.java:176)
 at android.app.ActivityThread.main(ActivityThread.java:6651)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:824)
Caused by: java.lang.IllegalArgumentException: Cannot add a null child view to a ViewGroup
 at android.view.ViewGroup.addView(ViewGroup.java:4703)
 at android.view.ViewGroup.addView(ViewGroup.java:4685)
 at uk.co.senab.actionbarpulltorefresh.library.PullToRefreshAttacher$DecorChildLayout.<init>(PullToRefreshAttacher.java:762)
 at uk.co.senab.actionbarpulltorefresh.library.PullToRefreshAttacher.<init>(PullToRefreshAttacher.java:139)
 at com.chanapps.four.activity.BoardActivity.bindPullToRefresh(BoardActivity.java:608)
 at com.chanapps.four.activity.BoardActivity.createAbsListView(BoardActivity.java:482)
 at com.chanapps.four.activity.BoardActivity.createViews(BoardActivity.java:250)
 at com.chanapps.four.activity.AbstractBoardSpinnerActivity.onCreate(AbstractBoardSpinnerActivity.java:84)
 at android.app.Activity.performCreate(Activity.java:7088)
 at android.app.Activity.performCreate(Activity.java:7079)
 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
 ... 10 more`
the next time i open the app it has the theme i selected before the error, so it does change the option before crashing

--- a/app/libs/actionBarPullToRefresh/src/main/java/uk/co/senab/actionbarpulltorefresh/library/PullToRefreshAttacher.java
+++ b/app/libs/actionBarPullToRefresh/src/main/java/uk/co/senab/actionbarpulltorefresh/library/PullToRefreshAttacher.java
@@ -756,10 +756,17 @@ public class PullToRefreshAttacher implements View.OnTouchListener {
             super(context);
 
             // Move all children from decor view to here
-            for (int i = 0, z = systemDecorView.getChildCount(); i < z; i++) {
+            int systemDecorViewChildCnt = systemDecorView.getChildCount();
+            for (int i = 0; i < systemDecorViewChildCnt; i++) {
                 View child = systemDecorView.getChildAt(i);
-                systemDecorView.removeView(child);
-                addView(child);
+
+                if (child != null) {
+                    systemDecorView.removeView(child);
+                    addView(child);
+                }
+
+                // cnt has to be updated because statements above may modify it
+                systemDecorViewChildCnt = systemDecorView.getChildCount();
             }
 
             /**

